src/analysis/souffle/dataflow_graph.dl

```diff
@@ -76,6 +77,25 @@
 // Rules
 //-----------------------------------------------------------------------------
 
+// Cross-populate operation_1 and operation.
+operation(owner, operator, result, [input1, nil]) :- operation_1(owner, operator, result, input1).
```

Comment:
If we can have operation as a list, why do we need `operation_x` variants?

Reply:
It's syntactic sugar to make writing tests and frontends easier. While `operation`s can, in general, have an arbitrary number of operations, we know that, in practice, they most often have 1 or 2 operands. Having a special syntax for that common case can provide a degree of parse-time checking for silly errors (such as using `*` with a single operand, for instance).

Reply:
I see. Call them `binary_operation` and `unary_operation` then? Where `operation` used in this PR, btw?

Reply:
Renaming done, now called `binaryOperation` and `unaryOperation`.

`operation` is used when populating the `operationToOperands` relation. This allows us to uniformly handle the default propagation rules and edge-drawing for any operation of any arity. See `arbitrary_operation_tag_propagation.dl`, where this is demonstrated for an `operation` with 4 inputs.